final-project/scripts/blog.js

```diff
@@ -0,0 +1,139 @@
+    }
+
+    async fetchBlogArticles () {
+        const {success, data} = await this.blogApi.get();
+        if (!success) {
+            return [];
+        }
+        return data;
+    }
+
+    moveSlider (target) {
+        const slides    = document.querySelectorAll('.blog__slider__slide'); // reference to the slides
+        const controls  = document.querySelectorAll('#blog__slider__controls li a'); // reference to the dots
+        
+        // Hide the current slide and display the new selected one
+        slides[this.currentSlide].classList.add('blog__slider__slide--hidden');
+        slides[target].classList.remove('blog__slider__slide--hidden'); 
+    
+        // Remove the selected class from the current dot and add it to the new selected one
+        controls.forEach(el => {
+            el.classList.remove('selectted');
+        });
+        controls[target].classList.add('selectted');
+        this.currentSlide = target;
+    }
+
+    nextSlide () {
+        const target = this.currentSlide < (this.totalSlides - 1) ? (parseInt(this.currentSlide) + 1) : 0;
+        this.moveSlider(target); 
+    }
+
+    previousSlide () {
+        const target = this.currentSlide > 0 ? (this.currentSlide - 1) : (this.totalSlides - 1);
+        this.moveSlider(target); 
+    }
+
+    swipeSlide () {
+        const swipeLength = this.touchStartX > 0 ? Math.abs(this.touchStartX - this.touchEndX) : 0;
+        if (swipeLength === 0 || swipeLength < 40) {
+            return;
+        }
+        if (this.touchStartX < this.touchEndX) {
+            this.previousSlide();
+        } else {
+            this.nextSlide();
+        }
+        this.touchStartX = 0;
+        this.touchEndX   = 0;
+    }
+
+    setSliderEventListeners () {
+        this.sliderControls.addEventListener('click', e => {
+            e.preventDefault();
+            if (e.target.tagName === 'A') {
+                this.moveSlider(e.target.getAttribute('data-slide'));
+            }
+        });
+
+        this.leftArrow.addEventListener('click', e => {
+            e.preventDefault();
+            this.previousSlide();
+        });
+
+        this.rightArrow.addEventListener('click', e => {
+            e.preventDefault();
+            this.nextSlide();
+        });
+
+        this.slidesContainer.addEventListener('touchstart', e => {
+            e.preventDefault();
+            const touch = e.touches.item(0);
+            this.touchStartX = touch.clientX;
+        });
+
+        this.slidesContainer.addEventListener('touchend', e => {
+            e.preventDefault();
+            const touch = e.changedTouches.item(0);
+            this.touchEndX = touch.clientX;
+            this.swipeSlide();
+        });
+    }
+
+    createSlides () {
+        let slides = '';
+        let controls = '';
+        this.articles.forEach((item, index) => {
+            slides += `<div class='blog__slider__slide ${index > 0 ? 'blog__slider__slide--hidden' : ''}'>
```

Comment:
In case that the HTML code for this changes, it could have been easier to "clone" the original node instead of duplicating it here, but this is ok for this challenge.